code/go/0chain.net/smartcontract/dbs/event/blobber_test.go

```diff
@@ -218,3 +219,46 @@ func TestBlobbers(t *testing.T) {
 	blobber, err = eventDb.GetBlobber(sn.ID)
 	require.Error(t, err)
 }
+
+func TestBlobberExists(t *testing.T) {
+	t.Skip("only for local debugging, requires local postgresql")
+	access := dbs.DbAccess{
+		Enabled:         true,
+		Name:            "events_db",
+		User:            os.Getenv("POSTGRES_USER"),
+		Password:        os.Getenv("POSTGRES_PASSWORD"),
+		Host:            os.Getenv("POSTGRES_HOST"),
+		Port:            os.Getenv("POSTGRES_PORT"),
+		MaxIdleConns:    100,
+		MaxOpenConns:    200,
+		ConnMaxLifetime: 20 * time.Second,
+	}
+	eventDb, err := NewEventDb(access)
+	require.NoError(t, err)
+	defer eventDb.Close()
+
+	err = eventDb.AutoMigrate()
+	require.NoError(t, err)
+	bl := Blobber{
+		BlobberID: "something",
+	}
+	res := eventDb.Store.Get().Create(&bl)
+	if res.Error != nil {
+		t.Errorf("Error while inserting blobber %v", bl)
+		return
+	}
+	gotExists, err := bl.exists(eventDb)
+
+	if !gotExists || err != nil {
+		t.Errorf("Exists function did not work want true got %v and err was %v", gotExists, err)
+	}
+	b2 := Blobber{
+		BlobberID: "somethingNew",
+	}
+	gotExists, err = b2.exists(eventDb)
+	if gotExists || err != nil {
```

Comment:
You're using older exist here.
I would suggest to make changes to exist function only and test the scenarios.

Reply:
yaa my bad forgot to do that was actually planning to that only 😓 
Np fixed that

Reply:
Done